ocaml/testsuite/tests/typing-layouts/annots.ml

```diff
@@ -343,6 +599,19 @@ Error: This field value has type 'b -> 'b which is less general than
 |}]
 (* CR layouts v1.5: that's a pretty awful error message *)
 
+let r = { field = fun (type a : value mod global) (x : a) -> x }
+[%%expect{|
+Line 1, characters 18-62:
+1 | let r = { field = fun (type a : value mod global) (x : a) -> x }
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: This field value has type 'b -> 'b which is less general than
```

Comment:
Another terrible error message

Reply:
In my mind, a good improvement that seems reasonable to implement would be:
```
type r_value = { field : 'a. 'a -> 'a }
let r = { field = fun (type a : immediate) (x : a) -> x }

[%%expect{|
type r_value = { field : 'a. 'a -> 'a; }
Line 2, characters 18-55:
2 | let r = { field = fun (type a : immediate) (x : a) -> x }
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This field value has type ('b: immediate). 'b -> 'b which is less general than
         'a. 'a -> 'a
       The kind of 'a is value
         because of the definition of r_value at line 1, characters 0-39.
       But the kind of 'a must be a subkind of immediate
         because of the annotation on the abstract type declaration for a.
|}]
```
(Note that `'b -> 'b` is now `('b : immediate). 'b -> 'b`).

Actually, I'm mildly surprised this doesn't already happen. This seems to correspond to case `C3` in `printtyp.ml`.

But I think this is outside the scope of this PR - this error message predates my earlier jkind PR.